src/escrow/EscrowHandler.sol

```diff
@@ -3,18 +3,32 @@ pragma solidity ^0.8.18;
 
 import "solmate/tokens/ERC721.sol";
 import "../types/OfferItems.sol";
+import {SafeTransferLib} from "solady/src/utils/SafeTransferLib.sol";
 
 abstract contract EscrowHandler is ERC721TokenReceiver {
     function _transferERC721(address collectionAddress, uint256 id, address from, address to) internal {
         ERC721 collection = ERC721(collectionAddress);
         collection.safeTransferFrom(from, to, id);
     }
 
-    // @dev function to transfer items from an offer type
+    function _transferERC20(address tokenAddress, uint256 amount, address from, address to) internal {
+        if (address(this) == from) {
+            SafeTransferLib.safeTransfer(tokenAddress, to, amount);
+        } else {
+            SafeTransferLib.safeTransferFrom(tokenAddress, from, to, amount);
+        }
+    }
+
+    // @dev function to transfer items from an offer type. We check for the amount to distinguish ERC20
```

Comment:
this won't work with ERC1155s though. Shouldn't we simply add the token type to offer items instead?

Reply:
I was thinking about this. you think that'd make the most sense?